oauth2_provider/oauth2_validators.py

```diff
@@ -665,3 +711,181 @@ def validate_refresh_token(self, refresh_token, client, request, *args, **kwargs
         # Temporary store RefreshToken instance to be reused by get_original_scopes and save_bearer_token.
         request.refresh_token_instance = rt
         return rt.application == client
+
+    @transaction.atomic
+    def _save_id_token(self, token, request, expires, *args, **kwargs):
+
+        scopes = request.scope or " ".join(request.scopes)
+
+        if request.grant_type == "client_credentials":
+            request.user = None
```

Comment:
I believe OIDC `client_credentials` flow doesn't support ID tokens, so I'm not sure how you would get here?